### PR TITLE
DAOS-5730 dfuse: Update dfuse help output.

### PR DESCRIPTION
**Pull request by ashleypittman**
Rework dfuse help so it's not incorrect and is a lot more verbose.
Add a --version option.
Update NLT to use the long options.
Make it so the -t option works.
Improve exit login in dfuse when it is unable to start.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: src/client/dfuse/dfuse_main.c [hunk 1]
```diff
@@ -217,21 +217,64 @@ dfuse_launch_fuse(struct dfuse_projection_info *fs_handle,
...
+		"If dfuse is running in background mode (the default unless launched via mpirun)\n"
+		"then it will stay in the foreground until the mount is registered with the\n"
+		"kernel to allow appropriate error reporting.\n"
+		"\n"
+		"Caching is on by default with short metadata timeouts and write-back data cache,\n"
+		"this can be disabled entirely for the mount by the use of command line options.\n"
+		"Further settings can be set on a per-container basis via the use of container\n"
+		"attributes.  If the --disable-caching option is given then no caching will be\n"
+		"performed and the container attributes are not used, if --disable-wb-cache is\n"
+		"given the data caching for the whole mount is performed in write-back mode and\n."
```
Comment (daltonbohning) [inline src/client/dfuse/dfuse_main.c:273]: Extra period
```suggestion
		"given the data caching for the whole mount is performed in write-back mode and\n"
```

File: src/client/dfuse/dfuse_main.c [hunk 2]
```diff
@@ -217,21 +217,64 @@ dfuse_launch_fuse(struct dfuse_projection_info *fs_handle,
...
-		"          --disable-caching    Disable all caching\n"
-		"          --disable-wb-cache   Use write-through rather than write-back cache\n",
-		name);
+		"	   --disable-caching	Disable all caching\n"
+		"	   --disable-wb-cache	Use write-through rather than write-back cache\n"
+		"\n"
+		"	-h --help		Show this help\n"
+		"	-v --version		Show version\n"
+		"\n"
+		"Specifying pool and container are optional, if not set then dfuse can connect to\n"
```
Comment (daltonbohning) [inline src/client/dfuse/dfuse_main.c:250]: Grammar nitpick: I don't believe it's "proper" to use a commas here (comma splicing). Generally, this should be two sentences or use a semicolon. Though, I'm not sure if this differs between American and British English.
```suggestion
		"Specifying pool and container are optional; if not set then dfuse can connect to\n"
```
```suggestion
		"Specifying pool and container are optional. If not set then dfuse can connect to\n"
```

File: src/client/dfuse/dfuse_main.c [hunk 3]
```diff
@@ -375,7 +424,7 @@ main(int argc, char **argv)
 		rc = dfuse_bg(dfuse_info);
 		if (rc != 0) {
 			printf("Failed to background\n");
-			return 2;
+			exit(2);
```
Comment (daltonbohning) [inline src/client/dfuse/dfuse_main.c:427]: Do we not want this to be a -DER code?

File: src/client/dfuse/dfuse_main.c [hunk 4]
```diff
@@ -217,21 +217,64 @@ dfuse_launch_fuse(struct dfuse_projection_info *fs_handle,
...
-		"	   --container=<name>	container UUID/label\n"
+		"	-m --mountpoint=<path>	Mount point to use\n"
+		"\n"
+		"	   --pool=label		pool UUID/label\n"
+		"	   --container=label	container UUID/label\n"
 		"	   --sys-name=STR	DAOS system name context for servers\n"
-		"	-S --singlethreaded	Single threaded\n"
-		"	-t --thread-count=COUNT Number of fuse threads to use\n"
+		"\n"
+		"	-S --singlethread	Single threaded\n"
```
Comment (daltonbohning) [inline src/client/dfuse/dfuse_main.c:241]: Is it safe to change `--singlethreaded` to `--singlethread`?
One such instance is:
https://github.com/daos-stack/daos/blob/cf0db8517c62c91dbd14079cd6e7220fb3848a7b/src/tests/ftest/util/dfuse_utils.py#L28
But it doesn't appear that anything else in the DAOS code base is using `--singlethread`

File: src/client/dfuse/dfuse_main.c (hunk 1 above)
Comment (ashleypittman) [reply src/client/dfuse/dfuse_main.c:273]: Done.

File: src/client/dfuse/dfuse_main.c (hunk 2 above)
Comment (ashleypittman) [reply src/client/dfuse/dfuse_main.c:250]: Done.

File: src/client/dfuse/dfuse_main.c (hunk 3 above)
Comment (ashleypittman) [reply src/client/dfuse/dfuse_main.c:427]: Possibly, although it's the exit status of the program and not logged so it probably doesn't make much difference.  The background code is both delicate and relatively small so I don't want to put too much support structure around it for corner-cases.  Failures here should be fairly localised and obvious.

Comment (daltonbohning) [reply src/client/dfuse/dfuse_main.c:427]: Makes sense!

**Review by daltonbohning — APPROVED** [review]

File: src/client/dfuse/dfuse_main.c [hunk 5]
```diff
@@ -217,21 +217,64 @@ dfuse_launch_fuse(struct dfuse_projection_info *fs_handle,
...
-		"          --disable-caching    Disable all caching\n"
-		"          --disable-wb-cache   Use write-through rather than write-back cache\n",
-		name);
+		"	   --disable-caching	Disable all caching\n"
+		"	   --disable-wb-cache	Use write-through rather than write-back cache\n"
+		"\n"
+		"	-h --help		Show this help\n"
+		"	-v --version		Show version\n"
+		"\n"
+		"Specifying pool and container are optional. If not set then dfuse can connect to\n"
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/client/dfuse/dfuse_main.c:250]: I need to write a man page as well, but everything here directly relates to command line options and their effects, so over time it should probably be in both.  I'm going to work on the github markup data next to document the caching options properly, then combine everything into a man page last.

[the remark this answers is not in the document]
Comment (mchaarawi) [reply src/client/dfuse/dfuse_main.c:250]: I just feel that this is too much just for a usage message.. anyway i leave it to you whether to keep it or not when you have the man page..

**Review by mchaarawi — APPROVED** [review]

**Pull request merged**